src/include/stir/IO/InputStreamWithRecords.inl

```diff
@@ -73,6 +73,9 @@ get_next_record(RecordT& record) const
   if (is_null_ptr(stream_ptr))
     return Succeeded::no;
 
+#ifdef STIR_OPENMP
+#pragma omp critical(LISTMODEIO)
+#endif
```

Comment:
this will just guard the next line (which could be the `assert` in debug-mode and the construction of `data_sptr` in release). That's probably not what you want. I think you need something like
```c++
#pragma omp critical(LISTMODEIO)
{
  stream operations here
}
```
annoying is that more than likely `stream_ptr->eof()` are not thread-safe (I didn't check) and you cannot `return` from within a critical region, so need to assign this to variables
```c++
bool stream_bad = false;

#pragma omp critical(LISTMODEIO)
{
  stream operations here
    if (stream_ptr->eof())
     stream_bad = true;
  else if (stream_ptr->bad())
    { 
      warning("Error after reading from list mode stream in get_next_record");
      stream_bad = true; 
    }
}
if (stream_bad)
  return Succeeded::no; 
```

Reply:
I see, this is a good idea